### PR TITLE
endpoint: Add DebugPolicy option

### PR DESCRIPTION
**Pull request by jrajahalme**
Add endpoint DebugPolicy option that, if enabled, logs endpoint policy
map update details to /var/run/cilium/state/endpoint-policy.log.

The new DebugPolicy option is enabled if the new flag
--debug-verbose=policy is set, but can be enabled also independently
via:

  cilium endpoint config <EPID> DebugPolicy=true

This feature was first developed in the v1.7 branch, so no 1.7 backport is needed.

Signed-off-by: Martynas Pumputis <m@lambda.lt>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

```release-note
Added new Cilium agent option --debug-verbose=policy to log policy map updates.
```

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]
Looks solid - hope you can use this to debug issues more easily in the future :-)

**Review by nebril — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Review by jibi — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]

**Comment by jrajahalme** [comment]
Adopted comment resolutions from the v1.7 branch (#14081)

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
retest-4.9

File: pkg/endpoint/log.go [hunk 1]
```diff
@@ -114,3 +139,61 @@ func (e *Endpoint) UpdateLogger(fields map[string]interface{}) {
 
 	atomic.StorePointer(&e.logger, unsafe.Pointer(l))
 }
+
+func (e *Endpoint) updatePolicyLogger(fields map[string]interface{}) {
```
Comment (kaworu) [inline pkg/endpoint/log.go:147]: The caller need to hold `Endpoint.mutex` for writing correct? If yes, can we add a comment please?

About `UpdateLogger` comment on holding the mutex, my understanding is that the caller must hold _`Endpoint.mutex`_ (not `Endpoint.Mutex`) for _writing_ (not for reading) even when `fields` is `nil` as we can reach the `atomic.StorePointer` at line 140.

Comment (jrajahalme) [reply pkg/endpoint/log.go:147]: I'll update the locking comments ;-)

**Comment by jrajahalme** [comment]
All tests passed before adding changes to comments only, no need to retest.

**Pull request merged**